packages/amplify_flutter/lib/config/amplify_config.dart

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:amplify_flutter/config/analytics/analytics_config.dart';
+import 'package:amplify_flutter/config/api/api_config.dart';
+import 'package:amplify_flutter/config/auth/auth_config.dart';
+import 'package:amplify_flutter/config/storage/storage_config.dart';
+
+part 'amplify_config.g.dart';
+
+class AmplifyConfig {
```

Comment:
I was thinking we leave the `JsonSerializable` annotations. Thoughts?

Reply:
Adding the annotations would have the benefit of easing re-running codegen when these configs get updated; conversely, it would pose the risk of adding friction for customers who are using conflicting versions of the codegen packages who would then have to handle dependency resolution. For now I think the risk outweighs the benefit, but if we find that these config values require frequent updates or we expand our usage of codegen (which we should!) then I would agree that the annotations should be committed.